### DISCUSSION TITLE
Shell.TitleView not allowing specific Page Titles

### DISCUSSION THREAD

**Discussion by BumpaRoy** [dotnet/maui#18194] [Q&A]
I am using a Shell.TitleView to display an icon in the title bar.  Works great.  But I also want to have the Page.TitleProperty of each page to have a specific value (Page One, Page Two, etc.) and I want to set this value in the Page ContentPage Title property.  I don't get any text displayed on title bar, but the image does.

**AppShell.xmal**
`<Shell
    x:Class="peMove.Maui.AppShell"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:TitleView="clr-namespace:peMove.Maui.Pages"
    xmlns:local="clr-namespace:peMove.Maui.Pages">

    <Shell.TitleView>
        <Image Source="recv_48_icon.png"
               HorizontalOptions="EndAndExpand"
               VerticalOptions="Center" />
    </Shell.TitleView>

    <Shell.FlyoutHeader>
        <Grid BackgroundColor="{StaticResource Secondary}">
            <Label Text="Modules" 
                   Margin="10"
                   TextColor="White"
                   FontSize="24"/>
        </Grid>
    </Shell.FlyoutHeader>

    <ShellContent Title="Lookup" Icon="lookup_48_icon.png" ContentTemplate="{DataTemplate local:MainPage}" />
    <ShellContent Title="Receiving" Icon="lookup_48_icon.png" ContentTemplate="{DataTemplate local:RECVPage}" />
    <ShellContent Title="Transfer" Icon="recv_128_icon.png" ContentTemplate="{DataTemplate local:XFERPage}" />

    <Shell.ItemTemplate>
        <DataTemplate>
            <Grid ColumnDefinitions="0.2*,0.8*">
                <Image Source="{Binding FlyoutIcon}"
                       Margin="5"
                       HeightRequest="45" />
                <Label Grid.Column="1"
                       Text="{Binding Title}"
                       Margin="5"
                       FontSize="24"
                       FontAttributes="Italic"
                       VerticalTextAlignment="Center" />
            </Grid>
        </DataTemplate>
    </Shell.ItemTemplate>
</Shell>`

**MainPage.xmal**
`<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:viewModels="clr-namespace:peMove.Maui.LookupViewModels"
             x:Class="peMove.Maui.Pages.MainPage"
             Title="Lookup">

    <ListView ItemsSource="{Binding Documents}" HasUnevenRows="True" VerticalOptions="FillAndExpand" RowHeight="-1">
        <ListView.BindingContext>
            <viewModels:ViewModel_L />
        </ListView.BindingContext>
`  ....

**App.xmal.cs**
`namespace peMove.Maui
{
    public partial class App : Application
    {
        public App()
        {
            InitializeComponent();
            MauiProgram.prefixes=LoadPrefixs();
            MainPage = new AppShell();
        }
`

**Comment by borrmann** [comment]
One way to get around this is by hiding the titleview entirely. I have had many issues and limitations with TitleView so I created my own Page and title Templates. 

The drawback is that you have to deal with adding an icon or something to click on for back navigation on pages that aren't root pages and that you might not be able to easily add the native controls for that if you require them.
It gives you a lot more design flexibility though and I don't mind using my own back navigation icons...
